### ISSUE TITLE
EC_POINT_cmp() on OpenSSL 1.1.1l returning 1

### ISSUE THREAD

**Issue by shaarang** [openssl/openssl#18333]
Hello,

For some strange reason EC_POINT_cmp() is returning 0 on OpenSSL 1.1.1l

And specifically this line returns zero:
int res = group->meth->point_cmp(group, a, b, ctx);

but I am not able to find any place in whole src code of OpenSSL 1.1.1l where point_cmp function ptr is set.
so I don't what function is actually getting called.

pls suggest.

**Comment by slontis** [comment]
Most methods like this just point to tables..
search for 
````
static const EC_METHOD
````
And you will find big tables of function pointers..

They normally use
   ossl_ec_GFp_simple_cmp
OR
  ossl_ec_GF2m_simple_cmp
  
  Using a debugger (gdb) to step into the function is the easiest way of figuring out what is happening.

**Comment by mattcaswell** [comment]
Are you expecting the comparison of the two points to indicate that they match, or that they don't match? A 0 return from that function is normal output - meaning the points match:

https://www.openssl.org/docs/man1.1.1/man3/EC_POINT_cmp.html

````
EC_POINT_cmp returns 1 if the points are not equal, 0 if they are, or -1 on error.
````

**Comment by shaarang** [comment]
Sorry I meant it is returning 1 and thus failing.

Actually I have traced down to this function form SSL_CTX_use_PrivateKey_file.
this function is working when I am using RSA certificate and pvt key but failing when I am trying to use ECC cert and key.

Which is strange for me because the same key (in PEM format) works for me in a separate program but not with the main big software.

While searching I find this which sounds similar but the difference is that I am not using any ENGINE:
https://github.com/openssl/openssl/issues/1532

**Comment by nhorman** [comment]
Marking as inactive, to be closed when 3.4 dev cycle ends, barring further input
